### PR TITLE
Add support for build config into oc set env

### PR DESCRIPTION
**Pull request by mfojtik**
Allow to run:

```console
[root@localhost origin]# oc set env bc/ruby-hello-world FOO=bar
buildconfig "ruby-hello-world" updated
[root@localhost origin]# oc set env bc/ruby-hello-world --list
# buildconfigs ruby-hello-world
FOO=bar
[root@localhost origin]# oc set env dc/ruby-hello-world bc/ruby-hello-world FOO=foo
deploymentconfig "ruby-hello-world" updated
buildconfig "ruby-hello-world" updated
```

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
@smarterclayton @kargakis PTAL

File: pkg/cmd/cli/cmd/set/env.go [hunk 1]
```diff
@@ -20,11 +20,13 @@ import (
 
 const (
 	envLong = `
-Update environment variables on a pod template
+Update environment variables on a pod template and build config
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/set/env.go]: or a build config

File: pkg/cmd/cli/cmd/set/env.go [hunk 2]
```diff
@@ -200,8 +202,23 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 			return nil
 		})
 		if !ok {
-			skipped++
-			continue
+			ok, err = f.UpdateObjectEnvironment(info.Object, func(vars *[]kapi.EnvVar) error {
```
Comment (kargakis) [inline pkg/cmd/cli/cmd/set/env.go]: I guess I am fine with two separate methods. @smarterclayton is there any use for this upstream?

Comment (mfojtik) [inline pkg/cmd/cli/cmd/set/env.go]: maybe I should add comment saying that this is a fallback function for things that don't have pod spec.

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
[test]

@smarterclayton bump.

**Comment by kargakis** [comment]
LGTM

File: pkg/cmd/cli/cmd/set/env.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/env.go]: The goal is to move set env upstream, so we will need it.

On Thu, Feb 25, 2016 at 12:12 PM, Michail Kargakis <notifications@github.com

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/env.go]: Yes

Comment (mfojtik) [inline pkg/cmd/cli/cmd/set/env.go]: @smarterclayton already added.

Comment (kargakis) [inline pkg/cmd/cli/cmd/set/env.go]: > The goal is to move set env upstream, so we will need it.

We definitely need it, I was asking if Kube had any use for it.

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/set/env.go]: Probably ConfigMap and Secrets.

**Comment by mfojtik** [comment]
@smarterclayton merge?

**Comment by smarterclayton** [comment]
Lgtm [merge]

**Pull request merged**